staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal_cbor.go

```diff
@@ -0,0 +1,35 @@
+package v1
+
+import (
+	"github.com/fxamacker/cbor/v2"
+)
+
+var _ cbor.Marshaler = JSONSchemaPropsOrBool{}
+var _ cbor.Unmarshaler = &JSONSchemaPropsOrBool{}
+
+var (
+	cborFalse = []byte{0xf4}
+	cborTrue  = []byte{0xf5}
+)
+
+func (s JSONSchemaPropsOrBool) MarshalCBOR() ([]byte, error) {
+	if s.Schema != nil {
+		return cbor.Marshal(s.Schema)
+	}
+
+	if s.Schema == nil && !s.Allows {
+		return cborFalse, nil
+	}
+	return cborTrue, nil
+}
+
+func (s *JSONSchemaPropsOrBool) UnmarshalCBOR(data []byte) error {
```

Comment:
It seems like a good idea to unmarshal into a temporary destination object as the json implementation does, since non-nil errors from Unmarshal don't guarantee that the destination object remains unmodified (it could have been partially filled up until the error was encountered).